Feature/smhe 2453 bij een retry de actuele ip gegevens gebruiken

natasja-n opened the pull request:
Add unit test for SchedulerTaskExecutorService to verify that the current ip address from the device is used to create a new request.

natasja-n committed:
SMHE-2453 Bij een retry de actuele ip gegevens gebruiken

Signed-off-by: natasjanortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-2453 Bij een retry de actuele ip gegevens gebruiken

Signed-off-by: natasjanortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-2453 Bij een retry de actuele ip gegevens gebruiken
update unit test DeviceResponseMessageServiceTest

Signed-off-by: natasjanortier <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' into feature/SMHE-2453_Bij_een_retry_de_actuele_ip_gegevens_gebruiken

jboon commented on osgp/platform/osgp-core/src/test/java/org/opensmartgridplatform/core/application/services/DeviceResponseMessageServiceTest.java:
```diff
@@ -167,4 +169,32 @@ public void testProcessScheduledMessageSuccess() {
+
+    this.deviceResponseMessageService.processMessage(message);
+
+    verify(this.deviceService, times(1))
```
Je test hier alleen of er een device opgehaald wordt, maar niet de logica die je er aan toegevoegd hebt. Je zou hier moeten testen of het networkAddress idd aan de ScheduledTask is toegevoegd. 
Als naam van de test methode verwacht ik iets wat lijkt op de story titel.

natasja-n replied:
ja klopt, maar die ScheduledTask heeft helemaal geen networkaddress of messagemetadata. Ik ga nog even verder zoeken, dit stond in de story maar vraag me af of dit de oplossing is.

natasja-n replied:
Nikki heeft getest en bevestigd dat het beschreven probleem opgelost is en niet meer bestaat. Ik heb nu alleen nog  in de ScheduledTaskExecutorService een extra unit test toegevoegd die checkt of bij het aanmaken van een request vanuit een ScheduledTask het ip adres vanuit het net opgehaalde device gebruikt wordt.

natasja-n committed:
SMHE-2453 Bij een retry de actuele ip gegevens gebruiken
scheduledTask is executed with latest device network address

Signed-off-by: natasjanortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-2453 Bij een retry de actuele ip gegevens gebruiken
scheduledTask is executed with latest device network address

Signed-off-by: natasjanortier <natasja.nortier@alliander.com>

harrymiddelburgetall commented on osgp/platform/osgp-core/src/test/java/org/opensmartgridplatform/core/application/tasks/ScheduledTaskExecutorServiceTest.java:
```diff
@@ -241,6 +241,50 @@ void testMetadataOfScheduledTaskToRetryRequest() throws FunctionalException {
+    when(this.scheduledTaskRepository.findByStatusAndScheduledTimeLessThan(
+            eq(ScheduledTaskStatusType.NEW), any(Timestamp.class), any(Pageable.class)))
+        .thenReturn(List.of(scheduledTask), Collections.emptyList());
+    when(this.scheduledTaskRepository.findByStatusAndScheduledTimeLessThan(
+            eq(ScheduledTaskStatusType.RETRY), any(Timestamp.class), any(Pageable.class)))
+        .thenReturn(new ArrayList<>());
```
This test does not check the device's IP-address in case of task in RETRY state. Is that on purpose?

natasja-n replied:
No, not on purpose. This test was just to prove that when a new request is created out of a scheduled task, the most recent ip address is being used. This code is the same, regardless of the ScheduledTaskStatusType. Updating it with both types will only add complexity and make it less understandable.

harrymiddelburgetall replied:
check

harrymiddelburgetall reviewed (changes requested):
Quite a complex unit test. Missing coverage on purpose?

natasja-n committed:
SMHE-2453 Bij_een_retry_de_actuele_ip_gegevens_gebruiken
update unit test testRequestFromScheduledTaskIsCreatedWithLatestDeviceNetworkAddress

Signed-off-by: natasjanortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-2453 Bij_een_retry_de_actuele_ip_gegevens_gebruiken
revert update unit test testRequestFromScheduledTaskIsCreatedWithLatestDeviceNetworkAddress

Signed-off-by: natasjanortier <natasja.nortier@alliander.com>
